src/calliope/backend/helper_functions.py

```diff
@@ -237,26 +240,15 @@ def as_array(self, input_component: str, *, over: str | list[str]) -> xr.DataArr
         Returns:
             xr.DataArray: resulting array.
         """
-        if input_component in self._input_data.data_vars:
-            component_da = self._input_data[input_component]
-            bool_component_da = (
-                component_da.notnull()
-                & (component_da != np.inf)
-                & (component_da != -np.inf)
+        if isinstance(input_component, str):
```

Comment:
This is all now dealt with before we reach here, by the `where` parsing evaluator.